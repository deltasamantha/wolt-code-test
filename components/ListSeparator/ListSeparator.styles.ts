import {StyleSheet} from "react-native";

import theme from "../../theme/theme";

export const styles = StyleSheet.create({
  rootContainer: {
    paddingHorizontal: theme.spacing.lg,
    backgroundColor: theme.colors.colorWhite,
  },
});
