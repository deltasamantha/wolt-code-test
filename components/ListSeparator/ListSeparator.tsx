import React from "react";
import {View} from "react-native";

import {styles} from "./ListSeparator.styles";
import Divider from "../Divider/Divider";

const ListSeparator: React.FC = () => {
  return (
    <View style={styles.rootContainer}>
      <Divider type="light" />
    </View>
  );
};

export default ListSeparator;
