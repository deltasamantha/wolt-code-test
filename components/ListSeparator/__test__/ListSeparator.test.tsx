import {render, cleanup, screen} from "@testing-library/react-native";

import ListSeparator from "../ListSeparator";

afterEach(cleanup);

describe("ListSeparator", () => {
  it("Should render ListHeader properly", () => {
    const {toJSON} = render(<ListSeparator />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Should render divider", () => {
    render(<ListSeparator />);
    const divider = screen.getByTestId("divider");
    expect(divider).toBeTruthy();
  });
});
