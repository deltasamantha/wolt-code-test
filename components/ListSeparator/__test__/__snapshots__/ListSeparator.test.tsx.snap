// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSeparator Should render ListHeader properly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      [
        {
          "height": 1,
          "width": "100%",
        },
        {
          "backgroundColor": "#EEEEEE",
        },
      ]
    }
    testID="divider"
  />
</View>
`;
